hack/lint.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+if ! which golangci-lint &>/dev/null; then
+  echo "Installing golangci-lint"
+  curl --retry 3 -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.25.1
+fi
```

Comment:
Oh please no. At least separate the installation from the execution. I definitely do not want to run random scripts from the internet when I run `make lint`. Make the script fail with suggesting how to install the tool maybe?

Reply:
I did it like that mostly to make the whole thing a self-contained change for CI. Do you prefer it to extend the existing lint job or do you prefer to have a new job for this?

Reply:
(This situation reminds me of why I built inrepoconfig...)

Reply:
I think it would be great to have it in the existing job.

Reply:
I guess I will go the dance of 1. Use the tool if present 2. Update image 3. Always use the tool then /shrug

Reply:
I mean our traffic in ci-tools is pretty low volume -- feel free to break us for a couple minutes while images build